controllers/provider-azure/pkg/webhook/controlplaneexposure/ensurer.go

```diff
@@ -58,6 +58,7 @@ func (e *ensurer) EnsureKubeAPIServerService(ctx context.Context, svc *corev1.Se
 	if err != nil {
 		return err
 	}
+	// TODO: get seed version instead of shoot version
```

Comment:
Remove the entire `comparedVersion` logic and put the annotation always. Leave a TODO to only do this for seeds with version >= 1.12 when getting the seed version becomes possible.